tooling-spi/src/main/java/org/opentest4j/reporting/tooling/spi/htmlreport/Block.java

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright 2021-2022 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.opentest4j.reporting.tooling.spi.htmlreport;
+
+import com.fasterxml.jackson.annotation.JsonTypeInfo;
+
+import static com.fasterxml.jackson.annotation.JsonTypeInfo.Id.SIMPLE_NAME;
+
+/**
+ * A block to be rendered in the HTML report.
+ *
+ * @param <T> the type of content
+ */
+@JsonTypeInfo(use = SIMPLE_NAME, property = "type")
+public interface Block<T> {
+
+	/**
+	 * {@return the content of this block}
+	 */
+	T getContent();
```

Comment:
What kinda of markup is supported here? Only plain text? Markdown? Arbitrary html?(probably a bad idea).